databroker/experimental/schemas.py

```diff
@@ -74,3 +65,90 @@ def is_mime_type(cls, v):
         if m_type not in mime_type_list:
             raise ValueError(f"{m_type} is not a valid mime type")
         return v
+
+
+class Document(BaseDocument):
+    data_blob: Optional[bytes]
+    data_url: Optional[pydantic.AnyUrl]
+
+    @pydantic.root_validator
+    def check_data_source(cls, values):
+        # Making them optional and setting default values might help to meet these conditions
+        # with the current data types without getting any conflicts
+        if values.get("data_blob") is not None and values.get("data_url") is not None:
+            raise ValueError(
+                "Not Valid: data_blob and data_url contain values. Use just one"
+            )
+        return values
+
+
+class DocumentRevision(BaseDocument):
+    revision: int
+
```

Comment:
It may be useful to add a classmethod constructor here to do what you were trying to do in `__init__`.

```py
@classmethod
def from_document(cls, document)
    return cls(key=document.key, ...)
```